### PR TITLE
[vxlan/vnet] Enhance VNET test to verify source VxLAN port for received encapsulated packets

### PR DESCRIPTION
**Pull request by ayurkiv-nvda**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
Add ability to verify UDP src port in VXLAN packet. Introduced new feature that allows to change range of possible UDP src port  in VXLAN packet, so need to extend existing test.

#### How did you do it?
Updated VNET test with the following logic:
- retrieve SAI_VXLAN_SRCPORT_RANGE_ENABLE from sai.profile
- define upper and lower bound for src port as test parameters
- check UDP src port in VXLAN  packet received from switch 

#### How did you verify/test it?
run:
```
py.test vxlan/test_vnet_vxlan.py --inventory "../ansible/inventory, ../ansible/veos" --host-pattern <testbed>-<topo> --module-path    ../ansible/library/ --testbed <testbed>-<topo>  --testbed_file ../ansible/testbed.csv    --allow_recover  --assert plain --log-cli-level info --show-capture=no -ra --showlocals --disable_loganalyzer --skip_sanity --upper_bound_udp_port 5555 --lower_bound_udp_port 1111
```
#### Any platform specific information?
according to these changes : [public PR hwsku]

#### Supported testbed topology if it's a new test case?
t0

### Documentation 

N/A

### CODE REVIEW

**Comment by volodymyrsamotiy** [comment]
Please change PR title to something like:
> [vxlan/vnet] Enhance VNET test to verify source VxLAN port for received encapsulated packets

File: ansible/roles/test/files/ptftests/vnet_vxlan.py [hunk 1]
```diff
@@ -39,6 +39,10 @@ def __init__(self):
         self.max_routes_wo_scaling = 1000
         self.vnet_batch = 8
         self.packets = []
+        self.vxlan_srcport_range_enabled = False
+        self.lower_bound_port = 0
+        self.upper_bound_port = 0
+
```
Comment (ayurkiv-nvda) [inline ansible/roles/test/files/ptftests/vnet_vxlan.py:45]: rename to : vxlan_srcport_upper_bound

File: ansible/roles/test/files/ptftests/vnet_vxlan.py [hunk 2]
```diff
@@ -307,13 +320,13 @@ def runTest(self):
 
         print
         for test in self.tests:
-            print test['name']
```
Comment (ayurkiv-nvda) [inline ansible/roles/test/files/ptftests/vnet_vxlan.py:310]: check if logging info also prints to standard output

File: ansible/roles/test/files/ptftests/vnet_vxlan.py [hunk 3]
```diff
@@ -470,7 +483,11 @@ def FromServer(self, test):
             logging.info(log_str)
 
             if not self.routes_removed:
-                verify_packet_any_port(self, masked_exp_pkt, self.net_ports)
+                received_pkt = verify_packet_any_port(self, masked_exp_pkt, self.net_ports)[1]
```
Comment (ayurkiv-nvda) [inline ansible/roles/test/files/ptftests/vnet_vxlan.py:486]: received_pkt = verify_packet_any_port(self, masked_exp_pkt, self.net_ports)[1] ->
status, received_pkt = verify_packet_any_port(self, masked_exp_pkt, self.net_ports)

File: ansible/roles/test/files/ptftests/vnet_vxlan.py [hunk 4]
```diff
@@ -470,7 +483,11 @@ def FromServer(self, test):
...
+                if self.vxlan_srcport_range_enabled:
+                    scapy_pkt  = Ether(received_pkt)
+                    assert self.lower_bound_port <= scapy_pkt.sport and self.upper_bound_port >=  scapy_pkt.sport, ("Received packet has UDP src port {} "
```
Comment (ayurkiv-nvda) [inline ansible/roles/test/files/ptftests/vnet_vxlan.py:489]: add brackets (self.lower_bound_port <= scapy_pkt.sport) and (self.upper_bound_port >=  scapy_pkt.sport)

File: ansible/roles/test/files/ptftests/vnet_vxlan.py [hunk 5]
```diff
@@ -470,7 +483,11 @@ def FromServer(self, test):
...
+                        "that is not beetween expected range {} - {}".format(scapy_pkt.sport, self.lower_bound_port, self.upper_bound_port))
```
Comment (ayurkiv-nvda) [inline ansible/roles/test/files/ptftests/vnet_vxlan.py:490]: that is not **in** expected range

File: tests/vxlan/conftest.py [hunk 6]
```diff
@@ -54,3 +54,19 @@ def pytest_addoption(parser):
         action="store_true",
         help="Apply new configurations on DUT"
     )
+
+    vxlan_group.addoption(
+        "--lower_bound_udp_port",
+        action="store",
+        default=0,
+        type=int,
+        help="Lowest expected src port for UPD packet"
```
Comment (ayurkiv-nvda) [inline tests/vxlan/conftest.py:63]: for vxlan udp packet

File: tests/vxlan/vnet_constants.py [hunk 7]
```diff
@@ -19,3 +19,5 @@
 NUM_ROUTES_KEY = "num_routes"
 NUM_ENDPOINTS_KEY = "num_endpoints"
 NUM_INTF_PER_VNET_KEY = "num_intf_per_vnet"
+LOWER_BOUND_UDP_PORT = "lower_bound_udp_port"
```
Comment (ayurkiv-nvda) [inline tests/vxlan/vnet_constants.py:22]: rename LOWER_BOUND_UDP_PORT -> LOWER_BOUND_UDP_PORT_KEY

File: tests/vxlan/test_vnet_vxlan.py [hunk 8]
```diff
@@ -11,6 +11,8 @@
 from tests.common.fixtures.ptfhost_utils import remove_ip_addresses, change_mac_addresses, \
                                                 copy_arp_responder_py, copy_ptftests_directory
 
+from vnet_constants import LOWER_BOUND_UDP_PORT, UPPER_BOUND_UDP_PORT
+
```
Comment (ayurkiv-nvda) [inline tests/vxlan/test_vnet_vxlan.py:15]: move import higher

File: tests/vxlan/test_vnet_vxlan.py [hunk 9]
```diff
@@ -21,6 +23,15 @@
 
 vlan_tagging_mode = ""
 
+def get_vxlan_srcport_range_enabled(duthost):
+    dut_platform = duthost.facts["platform"]
+    dut_hwsku = duthost.facts["hwsku"]
+    sai_profile = "/usr/share/sonic/device/%s/%s/sai.profile" % (dut_platform, dut_hwsku)
+    cmd = "cat {} | grep SAI_VXLAN_SRCPORT_RANGE_ENABLE | cut -d'=' -f2".format(sai_profile)
```
Comment (ayurkiv-nvda) [inline tests/vxlan/test_vnet_vxlan.py:30]: remove cat , use sai_profile after SAI_VXLAN_SRCPORT_RANGE_ENABLE

File: tests/vxlan/conftest.py [hunk 10]
```diff
@@ -54,3 +54,19 @@ def pytest_addoption(parser):
...
+    )
+
+    vxlan_group.addoption(
+        "--upper_bound_udp_port",
+        action="store",
+        default=255,
```
Comment (ayurkiv-nvda) [inline tests/vxlan/conftest.py:69]: change to max 2^16

File: ansible/roles/test/files/ptftests/vnet_vxlan.py (hunk 2 above)
Comment (ayurkiv-nvda) [reply ansible/roles/test/files/ptftests/vnet_vxlan.py]: not printed - revert

**Pull request closed without merging**